frontend/src/core/api/entity.js

```diff
@@ -125,7 +125,45 @@ export default class EntityAPI extends APIBase {
         const headers = new Headers();
         headers.append('X-Requested-With', 'XMLHttpRequest');
 
-        return await this.fetch('/get-history/', 'GET', payload, headers);
+        const results = await this.fetch('/get-history/', 'GET', payload, headers);
+
+        if (results.length === 0) {
+            return []
+        }
+
+        const keysMap = {
+            approved_user: 'approvedUser',
+            unapproved_user: 'unapprovedUser',
+            user_gravatar_url_small: 'userGravatarUrlSmall',
+            created_at: 'createdAt',
+            date_iso: 'dateIso',
+        };
+
+        const renameKeys = (keysMap, obj) => {
+            return Object
+                .keys(obj)
+                .reduce((acc, key) => {
+                    const renamedObject = {
+                        [keysMap[key] || key]: obj[key]
+                    };
+                    return {
+                        ...acc,
+                        ...renamedObject
+                    }
+                }, {});
+        };
+
+        const renamedResultsKeys = renameKeys(keysMap, results[0]);
+
+        const renamedCommentsKeys = results[0].comments.map(comment => {
+            return renameKeys(keysMap, comment);
+        });
+
+        if (renamedResultsKeys.comments) {
+            renamedResultsKeys.comments = renamedCommentsKeys;
+        }
+
+        return [renamedResultsKeys];
```

Comment:
I find this code too complex, and too specific. Once again, apologies for opening a can of worms here. Let's still get to the end of it, but by using a more generic solution that we can then apply to other places around the API.

I found this article that shows a nice solution to transform code from snake_case to camelCase: https://matthiashager.com/converting-snake-case-to-camel-case-object-keys-with-javascript

I'd take that code and put in the `api/base.js` file, add a method there to transform objects, and then use that method here to convert the results.